src/file.cr

```diff
@@ -377,6 +378,15 @@ class File < IO::FileDescriptor
     stat.size
   end
 
+  # Truncates the file to the specified length. Requires a write file descriptor
+  def truncate(length)
```

Comment:
I think that might be a bit hard to understand and unexpected for the default behaviour.  However I see its usefulness, how about doing that when passing `nil` or `true` for `length` (which should be called `size` btw)?